src/Symfony/Component/HttpClient/RetryableHttpClient.php

```diff
@@ -59,11 +62,18 @@ public function request(string $method, string $url, array $options = []): Respo
         $retryCount = 0;
         $content = '';
         $firstChunk = null;
+        $lastEventTime = microtime(true);
 
-        return new AsyncResponse($this->client, $method, $url, $options, function (ChunkInterface $chunk, AsyncContext $context) use ($method, $url, $options, &$retryCount, &$content, &$firstChunk) {
+        return new AsyncResponse($this->client, $method, $url, $options + ['timeout' => $this->retryTimeout], function (ChunkInterface $chunk, AsyncContext $context) use (&$lastEventTime, $method, $url, $options, &$retryCount, &$content, &$firstChunk) {
```

Comment:
If the user call the `request` method with an `timeout` option lower that `retryTimeout`, should'nt we use that timeout for the sub-request?

Reply:
I see what you want, we need a better solution. If you look at `CommonResponseTrait::initialize()`, you'll see that timeouts thrown during initialization are turned into errors. This is something that passthru filters don't have access to. This is what we should enhance instead. The initializer closure in AsyncResponse could ensure that an ErroChunk is passed to the passthru filter, so that the filter could act on it (dunno how exactly right now).

Reply:
Maybe that's all we need in this PR actually: be able to detect when a timeout is an error, and retry these. That would allow removing the retryTimeout argument.